src/uct/ib/base/ib_md.c

```diff
@@ -213,16 +213,31 @@ static ucs_status_t uct_ib_md_query(uct_md_h uct_md, uct_md_attr_t *md_attr)
     md_attr->cap.reg_mem_types = UCS_BIT(UCT_MD_MEM_TYPE_HOST);
 
     if (md->config.enable_gpudirect_rdma != UCS_NO) {
+        int enabled = 0;
+
         /* check if GDR driver is loaded */
         if (!access("/sys/kernel/mm/memory_peers/nv_mem/version", F_OK)) {
             md_attr->cap.reg_mem_types |= UCS_BIT(UCT_MD_MEM_TYPE_CUDA);
-            ucs_debug("%s: GPUDirect RDMA is enabled", uct_ib_device_name(&md->dev));
-        } else if (md->config.enable_gpudirect_rdma == UCS_YES) {
-            ucs_error("%s: Couldn't enable GPUDirect RDMA. Please make sure nv_peer_mem"
-                      " plugin installed correctly.", uct_ib_device_name(&md->dev));
-            return UCS_ERR_UNSUPPORTED;
-        } else {
-            ucs_debug("%s: GPUDirect RDMA is disabled", uct_ib_device_name(&md->dev));
+            ucs_debug("%s: CUDA GPUDirect RDMA is enabled", uct_ib_device_name(&md->dev));
+            enabled = 1;
+        }
+
+        /* check if ROCM KFD driver is loaded */
+        if (!access("/dev/kfd", F_OK)) {
+            md_attr->cap.reg_mem_types |= UCS_BIT(UCT_MD_MEM_TYPE_ROCM);
+            ucs_debug("%s: ROCM GPUDirect RDMA is enabled", uct_ib_device_name(&md->dev));
+            enabled = 1;
+        }
+
+        if (!enabled) {
+            if (md->config.enable_gpudirect_rdma == UCS_YES) {
+                ucs_error("%s: Couldn't enable GPUDirect RDMA. Please make sure"
```

Comment:
i would add a function like check_gpudirect_driver(md, sys_file_name, mem_type, debug_name) to handle both GPU/ROCm and print the failure if could not find

Reply:
but this failure have to be when both ROCM and CUDA is not present, so can't be put in a signal function for each GPU.

Reply:
i see, so this error message should still be common. only the ucs_debug() above should be inside the new function.
we can check here if  md_attr->cap.reg_mem_types has at least 1 non-host memtype